app-cpp/app355.cpp

```diff
@@ -393,6 +360,45 @@ void SatOrbit::OnSortOrbitalVectorMulti(orbit_iterator& inBegin, orbit_iterator&
     return inLHS.GetTLE().GetMeanMotion() < inRHS.GetTLE().GetMeanMotion();
 }
 
+void SatOrbit::CalculateOrbitalDataMulti(const std::vector<sat355::TLE>& inTLEVector, OrbitalDataVector& ioDataVector)
+{
+    // create a list of threads the size of the number of threads specified
+        std::vector<std::thread> threadVector{};
+        threadVector.reserve(mNumThreads);
+        const std::size_t tleVectorSize = inTLEVector.size();
+
+        // calculate the number of TLEs per thread
+        const std::size_t tlePerThread = tleVectorSize / mNumThreads;
+        auto tleBegin = inTLEVector.begin();
+        auto tleEnd = inTLEVector.end();
+
+        // loop through the threads
+        // (std::prtdiff_t is a signed version of std::size_t)
+        for (std::size_t i = 0; i < mNumThreads; ++i)
+        {
+            // calculate the begin and end of the TLEs for the current thread
+            tleBegin = inTLEVector.begin() + (i * tlePerThread);
+            tleEnd = tleBegin + tlePerThread;
+
+            // if this is the last thread, add the remaining TLEs
+            if (i == mNumThreads - 1)
+            {
+                tleEnd = inTLEVector.end();
+            }
```

Comment:
Use trinery operator `?:` to assign the value of `tleEnd`...
So that the assignment to `tleEnd` occurs in a single statement.
(Instead of this double-statement used currently to assign `tleEnd`)